driver-core/src/main/com/mongodb/client/model/geojson/GeometryCollection.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.geojson;
+
+import java.util.Collections;
+import java.util.List;
+
+import static com.mongodb.assertions.Assertions.isTrueArgument;
+import static com.mongodb.assertions.Assertions.notNull;
+
+/**
+ * A representation of a GeoJSON GeometryCollection.
+ *
+ * @since 3.1
+ */
+public final class GeometryCollection extends Geometry {
+    private final List<Geometry> geometries;
+
+    /**
+     * Construct an instance with the given list of Geometry objects
+     *
+     * @param geometries  the list of Geometry objects
+     */
+    public GeometryCollection(final List<Geometry> geometries) {
+        this(null, geometries);
+    }
+
+    /**
+     * Construct an instance with the given list of Geometry objects
+     *
+     * @param coordinateReferenceSystem the coordinate reference system
+     * @param geometries  the list of Geometry objects
+     */
+    public GeometryCollection(final CoordinateReferenceSystem coordinateReferenceSystem,
+                              final List<Geometry> geometries) {
+        super(coordinateReferenceSystem);
+        notNull("geometries", geometries);
+        isTrueArgument("geometrys contain only non-null elements", !geometries.contains(null));
```

Comment:
The problem with this wording is that, if the condition is false, the IllegalArgumentException's message will be: "state should be: geometrys cannot contain null values".  The way it is now, it will be: "state should be: geometrys contain only non-null elements"